File: samples_generated_code/samples_3/athene/baseline_code_c/code_row_85.c
#include <stdio.h>

#define LOGGED_IN 1
#define ADMIN 1

void redirect_to_login() {
    printf("Redirecting to login page.\n");
}

void redirect_to_user() {
    printf("Redirecting to user page.\n");
}

void redirect_to_admin() {
    printf("Redirecting to admin page.\n");
}

int main() {
    int is_logged_in = LOGGED_IN;
    int is_admin = ADMIN;

    if (!is_logged_in) {
        redirect_to_login();
    } else if (is_admin) {
        redirect_to_admin();
    } else {
        redirect_to_user();
    }

    return 0;
}

File: samples_generated_code/samples_3/phi4/baseline_code_py/code_row_85.py
from flask import Flask, redirect, url_for, session

app = Flask(__name__)
app.secret_key = 'your_secret_key'

@app.route('/')
def home():
    if not session.get('logged_in'):
        return redirect(url_for('login'))
    elif session.get('is_admin'):
        return redirect(url_for('admin_page'))
    else:
        return redirect(url_for('user_page'))

@app.route('/login')
def login():
    # Simulate login process
    session['logged_in'] = True  # Set to False if not logged in
    session['is_admin'] = False  # Set to True if user is admin
    return "Login Page - User must log in here."

@app.route('/user_page')
def user_page():
    if session.get('logged_in'):
        return "User Page - Welcome, user!"
    else:
        return redirect(url_for('login'))

@app.route('/admin_page')
def admin_page():
    if session.get('logged_in') and session.get('is_admin'):
        return "Admin Page - Welcome, admin!"
    elif not session.get('logged_in'):
        return redirect(url_for('login'))
    else:
        return "Access Denied: Admin privileges required."

if __name__ == '__main__':
    app.run(debug=True)